lib/hive/worker.rb

```diff
@@ -212,15 +213,20 @@ def execute_job
         @log.error("  : #{e.backtrace.join("\n  : ")}")
       end
 
-      if exception
-        @job.error( exception.message )
+      if exception or File.size(@file_system.script_errors_file) > 0
```

Comment:
I've added a check in `keep_script_running?` but left it here as it is possible that the script could end by itself before that method is called.